File: rb/features.rb
# NhlApiDocumentation SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module NhlApiDocumentationFeatures
  def self.make_feature(name)
    case name
    when "base"
      NhlApiDocumentationBaseFeature.new
    when "test"
      NhlApiDocumentationTestFeature.new
    else
      NhlApiDocumentationBaseFeature.new
    end
  end
end

File: py/nhlapidocumentation_sdk/features.py
# NhlApiDocumentation SDK feature factory

from nhlapidocumentation_sdk.feature.base_feature import NhlApiDocumentationBaseFeature
from nhlapidocumentation_sdk.feature.test_feature import NhlApiDocumentationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NhlApiDocumentationBaseFeature(),
        "test": lambda: NhlApiDocumentationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
